rsocket-micrometer/src/main/java/io/rsocket/micrometer/MicrometerDuplexConnection.java

```diff
@@ -0,0 +1,244 @@
+  }
+
+  @Override
+  public void dispose() {
+    delegate.dispose();
+  }
+
+  @Override
+  public Mono<Void> onClose() {
+    return delegate.onClose();
+  }
+
+  @Override
+  public Flux<Frame> receive() {
+    return delegate.receive().doOnNext(frameCounters);
+  }
+
+  @Override
+  public Mono<Void> send(Publisher<Frame> frames) {
+    Objects.requireNonNull(frames, "frames must not be null");
+
+    return delegate.send(Flux.from(frames).doOnNext(frameCounters));
+  }
+
+  private static final class FrameCounters implements Consumer<Frame> {
+
+    private final Logger logger = LoggerFactory.getLogger(this.getClass());
+
+    private final Counter cancel;
+
+    private final Counter complete;
+
+    private final Counter error;
+
+    private final Counter extension;
+
+    private final Counter keepalive;
+
+    private final Counter lease;
+
+    private final Counter metadataPush;
+
+    private final Counter next;
+
+    private final Counter nextComplete;
+
+    private final Counter requestChannel;
+
+    private final Counter requestFireAndForget;
+
+    private final Counter requestN;
+
+    private final Counter requestResponse;
+
+    private final Counter requestStream;
+
+    private final Counter resume;
+
+    private final Counter resumeOk;
+
+    private final Counter setup;
+
+    private FrameCounters(Type connectionType, MeterRegistry meterRegistry, Tag... tags) {
+      this.cancel = counter(connectionType, meterRegistry, CANCEL, tags);
+      this.complete = counter(connectionType, meterRegistry, COMPLETE, tags);
+      this.error = counter(connectionType, meterRegistry, ERROR, tags);
+      this.extension = counter(connectionType, meterRegistry, EXT, tags);
+      this.keepalive = counter(connectionType, meterRegistry, KEEPALIVE, tags);
+      this.lease = counter(connectionType, meterRegistry, LEASE, tags);
+      this.metadataPush = counter(connectionType, meterRegistry, METADATA_PUSH, tags);
+      this.next = counter(connectionType, meterRegistry, NEXT, tags);
+      this.nextComplete = counter(connectionType, meterRegistry, NEXT_COMPLETE, tags);
+      this.requestChannel = counter(connectionType, meterRegistry, REQUEST_CHANNEL, tags);
+      this.requestFireAndForget = counter(connectionType, meterRegistry, REQUEST_FNF, tags);
+      this.requestN = counter(connectionType, meterRegistry, REQUEST_N, tags);
+      this.requestResponse = counter(connectionType, meterRegistry, REQUEST_RESPONSE, tags);
+      this.requestStream = counter(connectionType, meterRegistry, REQUEST_STREAM, tags);
+      this.resume = counter(connectionType, meterRegistry, RESUME, tags);
+      this.resumeOk = counter(connectionType, meterRegistry, RESUME_OK, tags);
+      this.setup = counter(connectionType, meterRegistry, SETUP, tags);
+    }
+
+    @Override
+    public void accept(Frame frame) {
+      FrameType frameType = frame.getType();
+
+      switch (frameType) {
+        case RESERVED:
```

Comment:
Can we let this fall to the default case?

Reply:
Yep, same oversight as `PAYLOAD`, letting the IDE get the better of me.